controllers/JobController.js

```diff
@@ -67,59 +25,80 @@ exports.JobListByApplicantId = [
  * @returns {Object}
  */
 exports.JobList = [
-	auth,
-	function (req, res) {
-		try {
-			JobModel.find().then((Jobs) => {
-				if (Jobs.length > 0) {
-					let JobData = [];
-					Jobs.forEach(job => {
-						let data = new jobData(job);
-						JobData.push(data);
-					});
-					return apiResponse.successResponseWithData(res, "Operation success", JobData);
-				} else {
-					return apiResponse.successResponseWithData(res, "Operation success", []);
-				}
-			});
-		} catch (err) {
-			//throw error in json response with status 500.
-			return apiResponse.ErrorResponse(res, err);
-		}
-	}
+  auth,
+  body("postedBy", "postedBy must be a String").isString(),
+  function (req, res) {
+    try {
+      const filterObj = {
+        isDeleted: false,
+      };
+      const body = req.body;
+
+      if (body.postedBy) {
+        filterObj.postedBy = body.postedBy;
+      }
+
+      JobModel.find(filterObj)
+        .sort("-createdAt")
+        .then((Jobs) => {
+          if (Jobs.length > 0) {
+            let JobData = [];
+            Jobs.forEach((job) => {
+              let data = new jobData(job);
+              JobData.push(data);
+            });
+            return apiResponse.successResponseWithData(
+              res,
+              "Operation success",
+              JobData
+            );
+          } else {
+            return apiResponse.successResponseWithData(
+              res,
+              "Operation success",
+              []
+            );
+          }
+        });
+    } catch (err) {
+      //throw error in json response with status 500.
+      return apiResponse.ErrorResponse(res, err);
+    }
+  },
 ];
 
 /**
- * Job Details by email.
+ * Single Job.
  *
- * @param {string}      email
+ * @param {string}      id
  *
  * @returns {Object}
  */
-exports.JobsByEmail = [
-	auth,
-	function (req, res) {
-		if (!req.params.email) {
-			return apiResponse.successResponseWithData(res, "Operation success", []);
-		}
-		try {
-			JobModel.find({ email: req.params.email }).then((Jobs) => {
-				if (Jobs.length > 0) {
-					let JobData = [];
-					Jobs.forEach(job => {
-						let data = new jobData(job);
-						JobData.push(data);
-					});
-					return apiResponse.successResponseWithData(res, "Operation success", JobData);
-				} else {
-					return apiResponse.successResponseWithData(res, "Operation success", []);
-				}
-			});
-		} catch (err) {
-			//throw error in json response with status 500.
-			return apiResponse.ErrorResponse(res, err);
-		}
-	}
+exports.JobById = [
+  auth,
+  // body("id", "Id must not be empty.").notEmpty(),
+  function (req, res) {
+    try {
+      JobModel.findOne({ _id: req.body.id }, (error, job) => {
+        if (error) {
+          return apiResponse.ErrorResponse(res, "Operation Failed.", error);
+        }
+
+        if (job) {
+          return apiResponse.successResponseWithData(
+            res,
+            "Operation Succeed.",
```

Comment:
Codacy has a fix for the issue: [Expected indentation of 6 tabs but found 12 spaces.](https://app.codacy.com/gh/n4alphapesto/hirebook-backend/pullRequest?prid=7771833)
```suggestion
						"Operation Succeed.",
```